### PR TITLE
Feature fixes

### PR DESCRIPTION
**Pull request by AntiAki-M**
## Description :sparkles:
Bug fixes

### Closes :no_good_woman:
**[ATV-86](https://helsinkisolutionoffice.atlassian.net/browse/ATV-86): Add support for Service actors for audit log** 
**[ATV-109](https://helsinkisolutionoffice.atlassian.net/browse/ATV-109): Status updates put new history on every PATCH** 
**[ATV-113](https://helsinkisolutionoffice.atlassian.net/browse/ATV-113): Accessing Document metadata API without authentication results in an error** 

### Automated tests :gear:️
Yes

### CODE REVIEW

File: documents/api/viewsets.py [hunk 1]
```diff
@@ -57,6 +63,8 @@ class DocumentMetadataViewSet(AuditLoggingModelViewSet):
 
     def get_queryset(self):
         user = self.request.user
+        if user.is_anonymous:
```
Comment (vinikvuo) [inline documents/api/viewsets.py:66]: This could be done be IsAuthenticated permissions?

File: documents/api/viewsets.py [hunk 2]
```diff
@@ -316,18 +324,23 @@ def partial_update(self, request, pk, *args, **kwargs):
             attachment_serializer = CreateAttachmentSerializer(data=data)
             attachment_serializer.is_valid(raise_exception=True)
             attachment_serializer.save()
-        # Don't update history if patch has no data.
-        if request.data:
+        # Update history only if status changed.
+        if request.data.get("status"):
```
Comment (vinikvuo) [inline documents/api/viewsets.py:328]: you could write here request.data.get("status") is not None:

File: documents/api/viewsets.py [hunk 3]
```diff
@@ -316,18 +324,23 @@ def partial_update(self, request, pk, *args, **kwargs):
...
                     "value": document.status,
-                    "timestamp": document.updated_at,
+                    "timestamp": document.status_timestamp,
                 }
             )
             status_history_serializer.is_valid(raise_exception=True)
             status_history_serializer.save()
 
+            # Make sure the request data query dict is mutable, assign status_timestamp field to be updated.
+            request.data._mutable = True
```
Comment (vinikvuo) [inline documents/api/viewsets.py:340]: What are you doing here?

File: documents/serializers/document.py [hunk 4]
```diff
@@ -18,9 +18,10 @@
 
 
 def status_to_representation(representation):
+    status_timestamp = representation.pop("status_timestamp")
```
Comment (vinikvuo) [inline documents/serializers/document.py:21]: Why here is this pop?

Comment (AntiAki-M) [reply documents/serializers/document.py:21]: By default the status timestamp would be shown as an individual field in the response, but here it's used as a part of status structure. Pop is used to remove the key and value from the original representation, and its value is moved into status structure.

File: documents/api/viewsets.py (hunk 3 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:340]: In some situations request.data isn't mutable by default. The purpose here is to allow adding a new field to be saved with the serializer. If you have any alternative suggestions for this I would be glad to know.

File: documents/api/viewsets.py (hunk 2 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:328]: That is true. Would it matter?

Comment (vinikvuo) [reply documents/api/viewsets.py:328]: It could be clearer and more excplicit.

File: documents/api/viewsets.py (hunk 3 above)
Comment (vinikvuo) [reply documents/api/viewsets.py:340]: I would insert timezone.now() in Serializers update method straight to the right field

File: documents/api/viewsets.py (hunk 1 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:66]: I used this solution as it matches the code style elsewhere in the project.

**Pull request merged**